features/steps/test_behave.py

```diff
@@ -1,108 +1,229 @@
-  if len(candidates) == 0:
-    print("No candidates for: '{0}'".format(pkg))
-  assert len(candidates) == 1
-  return candidates[0]
+    """ Find the package file in the repository """
+    candidates = glob.glob('/repo/' + pkg + '*.rpm')
+    if len(candidates) == 0:
+        print("No candidates for: '{0}'".format(pkg))
+    assert len(candidates) == 1
+    return candidates[0]
+
 
 def decorate_rpm_packages(pkgs):
-  " Converts package names like TestA, TestB into absolute paths "
-  return [find_pkg(p) for p in pkgs]
+    """ Converts package names like TestA, TestB into absolute paths """
+    return [find_pkg(p) for p in pkgs]
+
 
 def get_package_list():
-  " Gets all installed packages in the system "
-  pkgstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}\n'])
-  return pkgstr.splitlines()
+    """ Gets all installed packages in the system """
+    pkgstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}\n'])
+    return pkgstr.splitlines()
+
+
+def get_package_version_list():
+    """ Gets all installed packages in the system with version"""
+    pkgverstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}-%{VERSION}-%{RELEASE}\n'])
+    return pkgverstr.splitlines()
+
+
+def get_package_dnf_version_list():
+    """ Gets all installed packages in the system with version to check that dnf has same data about installed packages"""
+    pkgverstr = subprocess.check_output(['dnf', 'repoquery', '--installed', '-C', '--queryformat', '%{name}.%{version}.%{release}\n'])
+    pkgverstr = pkgverstr.splitlines()
+    return pkgverstr[1:]
+
+
 
 def diff_package_lists(a, b):
-  " Computes both left/right diff between lists `a` and `b` "
-  sa, sb = set(a), set(b)
-  return (map(_left_decorator, list(sa - sb)),
-      map(_right_decorator, list(sb - sa)))
+    """ Computes both left/right diff between lists `a` and `b` """
+    sa, sb = set(a), set(b)
+    return (map(_left_decorator, list(sa - sb)),
+        map(_right_decorator, list(sb - sa)))
+
+
+def package_version_lists(pkg, list_ver):
+    """ Select package versions """
+    found_pkgs = [x for x in list_ver if re.search('^' + pkg, x)]
+    assert len(found_pkgs) == 1
+    return str(found_pkgs[0])
+
+
+def package_absence(pkg, list_ver):
+    """ Select package versions """
+    found_pkgs = [x for x in list_ver if re.search('^' + pkg, x)]
+    assert len(found_pkgs) == 0
+    return None
+
 
 def execute_dnf_command(cmd, reponame):
-  " Execute DNF command with default flags and the specified `reponame` enabled "
-  flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
-  return subprocess.check_call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+    """ Execute DNF command with default flags and the specified `reponame` enabled """
+    flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
+    return subprocess.check_call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+
+
+def execute_dnf_command_notinstall(cmd, reponame):
```

Comment:
I had objection to `execute_dnf_command` and `execute_dnf_command_notinstall` functions. Reuse this if needed https://github.com/shaded-enmity/richdeps-docker/commit/9a70fd3edfe06ea71e908e3bd62193fef1c25e63